### ISSUE TITLE
Brightness slider only starts reducing monitor brightness below 50% with Apple Cinema Display 24" (Mini DisplayPort)

### ISSUE THREAD

**Issue by ArkanaDaze** [MonitorControl/MonitorControl#939]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the bug

When I use the brightness slider in MonitorControl to reduce the brightness of my Apple Cinema Display 24" (Mini DisplayPort) it only actually starts reducing brightness when it goes below 50%. This is also reflected in the OSD info I get. 
With the slider between 50% and 100% nothing happens, but when it goes below 50% it then works as expected.

### Steps to reproduce

Just use the brightness slider in the App, but I suppose you will need to use an (old) Apple Cinema Display 24" (Mini DisplayPort) in order to reproduce it... :-)

### Expected behavior

Brightness slider should work in gradually between 0% and 100%, and not only between 0 and 50%.

### Environment Information (please complete the following information)

```markdown
- macOS version: Monterey 12.2
- Mac model: MacBook Pro (14-inch, 2021) with M1 Max CPU
- MonitorControl version: 4.0.2
- Monitor(s): pple Cinema Display 24" (Mini DisplayPort)
- Apple Silicon/M1 (yes or no): Yes
```

**Comment by ArkanaDaze** [comment]
Actually solved by looking at the inputs provided by waydabber in #739. If I disable hardware DDC and just keep "Combine hardware and software dimming" enable it works ok from 05 to 100%. So this means I do not seem to have hardware DDC working anymore with my new M1 system, whereas it was still working ok when I was using an Intel MacBook Pro 13" with Monterey 12.2.

**Comment by the0neyouseek** [comment]
Hi @ArkanaDaze,
Please have a look here as well : https://github.com/MonitorControl/MonitorControl/discussions/896 has it seems this problem is quite current with apple displays

**Comment by ArkanaDaze** [comment]
Hi @the0neyouseek, yes, I had also checked #896 but it didn´t address my issues. What´s really working in my case is disabling "Use hardware DDC control" and then it seems to be working fine from 0% to 100% brightness via software dimming. In this case, it makes no difference if I have "Combine hardware and software dimming" enabled or not, as I am always to use the Apple external keyboard brightness buttons without issues.
Additionally, if I have both "Use hardware DDC control" and "Combine hardware and software dimming" enabled it also works, but the slider will only change monitor brightness between 0% and 50% range. Nevertheless, within this range it still seems to be providing me with full brightness when the slider is at 50%. Weird behaviour...

**Comment by waydabber** [comment]
Hi @ArkanaDaze - this is because when `Combine...` is neabled, brightness levels from 0-50% will be software dimming while DDC brightness will be at 0% in this range. From 51-100% DDC brightness change will kick in. This is to blend the two ways of brightness control to give an increased range of dark->bright spectrum (you can change the transition point of software vs hardware brightness in the advanced settings of the Displays tab.). On your config DDC control does not seem to work so from 50% to 100% you don't see any changes.

**Comment by ArkanaDaze** [comment]
Hi @waydabber, oh ok, I checked the Advanced Settings and now understand what you mean! You have a nice combined settings switchover point there that´s currently set for 50%... :-)
Anyway, it´s working now via software brightness, even though the range might not be as flexible as if I were able to combine software dimming with DDC. MonitorControl saved my day!

**Comment by waydabber** [comment]
Yeah, if DDC does not work, software (gamma) control is a fine alternative - in some aspects it might be even better than DDC. It is interesting though that DDC woked on Intel but stopped working on M1 with this display.

**Comment by ArkanaDaze** [comment]
Yes @waydabber I agree with you software control is a fine alternative. I also find it weird that this stopped working with the M1 MBPro, but since Apple changed extensively the way they work with and support external displays in the M1 architecture, maybe my problem was to be expected...

**Comment by emilgpa** [comment]
Same problem here @waydabber but with Huawei Mateview instead of Apple monitor. I can confirm that with imac 27 5K intel this issue not occurs but yes with mac M1.

**Comment by waydabber** [comment]
Hi @emilgpa - what connection are you using and which mac model do you have? Thanks!

**Comment by emilgpa** [comment]
Hi again @waydabber. connected by HDMI 2.0 and mac studio m1 max.

**Comment by waydabber** [comment]
Oh yes, that's the problem. Unfortunately the new Mac Studio (however amazing machine it is) still has the same old HDMI converter chip inside as the 2018 Intel and 2020 M1 mini or the 2021 macBook Pro - it simply blocks all DDC communication. You should try using an USB-C to DisplayPort cable. Some USB-C to HDMI dongles will work also work (if they convert ddc i2c communication properly - some fail at doing this properly)

See this for more info: https://github.com/MonitorControl/MonitorControl/discussions/750
